File: backend/internal/cli/project.go
package cli

import (
	"context"
	"fmt"
	"strings"
)

func (e commandEnv) runProject(ctx context.Context, args []string) int {
	if len(args) == 0 {
		return e.fail("missing project command")
	}
	if args[0] == "create" {
		return e.createProject(ctx, args[1:])
	}
	return e.fail("unknown project command: " + args[0])
}

func (e commandEnv) createProject(ctx context.Context, args []string) int {
	fs := newFlagSet("project create")
	addr, asJSON := e.addCommonFlags(fs)
	id := fs.String("id", "", "project id")
	name := fs.String("name", "", "project name")
	cwd := fs.String("cwd", "", "project working directory")
	autoRun := fs.Bool("auto-run", false, "start runner automatically")
	model := fs.String("model", "", "Codex model")
	approval := fs.String("approval-policy", "never", "Codex approval policy")
	sandbox := fs.String("sandbox", "workspace-write", "Codex sandbox mode")
	if err := fs.Parse(args); err != nil {
		return e.fail(err.Error())
	}
	ctx = withFlagToken(ctx, fs)
	payload, err := projectPayload(*id, *name, *cwd, *autoRun, *model, *approval, *sandbox)
	if err != nil {
		return e.fail(err.Error())
	}
	var project projectDTO
	if err := postJSON(ctx, e.client, *addr, "/api/projects", payload, &project); err != nil {
		return e.fail(err.Error())
	}
	if err := writeProject(e.out, project, *asJSON); err != nil {
		return e.fail(err.Error())
	}
	return 0
}

func projectPayload(id, name, cwd string, autoRun bool, model string, approval string, sandbox string) (projectDTO, error) {
	project := projectDTO{ID: strings.TrimSpace(id), Name: strings.TrimSpace(name), CWD: strings.TrimSpace(cwd),
		Model: strings.TrimSpace(model), ApprovalPolicy: strings.TrimSpace(approval), Sandbox: strings.TrimSpace(sandbox)}
	if autoRun {
		project.AutoRun = 1
	}
	if project.ID == "" {
		return project, fmt.Errorf("--id is required")
	}
	if project.CWD == "" {
		return project, fmt.Errorf("--cwd is required")
	}
	return project, nil
}

File: backend-ts/src/cli/project.ts
import { parseCommandArgs } from "./common.ts";
import { postJSON } from "./http.ts";
import { formatProject } from "./output.ts";
import type { EnvReader, Fetcher, ProjectDTO } from "./types.ts";

const PROJECT_CREATE_FLAGS = [
  { name: "id", required: true },
  { name: "name" },
  { name: "cwd", required: true },
  { name: "provider" },
  { name: "default-agent-profile" },
  { name: "model" },
  { name: "approval-policy" },
  { name: "sandbox" }
] as const;

export async function runProject(args: string[], env: EnvReader, fetcher: Fetcher): Promise<string> {
  const command = args[0]?.trim();
  if (!command) throw new Error("missing project command");
  if (command === "create") return await createProject(args.slice(1), env, fetcher);
  throw new Error(`unknown project command: ${command}`);
}

async function createProject(args: string[], env: EnvReader, fetcher: Fetcher): Promise<string> {
  const { common, values } = parseCommandArgs(args, [...PROJECT_CREATE_FLAGS], env);
  const project = await postJSON<ProjectDTO>(fetcher, common, "/api/projects", projectPayload(values));
  return formatProject(project, common.json);
}

function projectPayload(values: Record<string, string>): Record<string, unknown> {
  return {
    id: values.id,
    name: values.name ?? "",
    cwd: values.cwd,
    provider: values.provider ?? "codex",
    default_agent_profile_id: values["default-agent-profile"] ?? "",
    model: values.model ?? "",
    approval_policy: values["approval-policy"] ?? "never",
    sandbox: values.sandbox ?? "workspace-write"
  };
}
